src/money.c

```diff
@@ -130,23 +131,33 @@ void SubtractMoneyFromVar0x8005(void)
 
 void PrintMoneyAmountInMoneyBox(u8 windowId, int amount, u8 speed)
 {
-    PrintMoneyAmount(windowId, 38, 1, amount, speed);
+    PrintMoneyAmount(windowId, CalculateMoneyTextHorizontalPosition(amount), 1, amount, speed);
+}
+
+static u32 CalculateLeadingSpacesForMoney(u32 numDigits)
+{
+    u32 leadingSpaces = CountDigits(INT_MAX) - StringLength(gStringVar1);
+    return (numDigits > 8) ? leadingSpaces : leadingSpaces - 2;
 }
 
 void PrintMoneyAmount(u8 windowId, u8 x, u8 y, int amount, u8 speed)
 {
-    u8 *txtPtr;
-    s32 strLength;
+    u8 *txtPtr = gStringVar4;
+    u32 numDigits = CountDigits(amount);
+    u32 maxDigits = (numDigits > 6) ? MAX_MONEY_DIGITS: 6;
+    u32 leadingSpaces;
 
-    ConvertIntToDecimalStringN(gStringVar1, amount, STR_CONV_MODE_LEFT_ALIGN, 6);
+    ConvertIntToDecimalStringN(gStringVar1, amount, STR_CONV_MODE_LEFT_ALIGN, maxDigits);
 
-    strLength = 6 - StringLength(gStringVar1);
-    txtPtr = gStringVar4;
+    leadingSpaces = CalculateLeadingSpacesForMoney(numDigits);
 
-    while (strLength-- > 0)
+    while (leadingSpaces-- > 0)
         *(txtPtr++) = CHAR_SPACER;
 
     StringExpandPlaceholders(txtPtr, gText_PokedollarVar1);
+
+    if (numDigits > 8)
+        PrependFontIdToFit(gStringVar4, txtPtr + 1 + numDigits, FONT_NORMAL, 54);
```

Comment:
Initially I was going to say that `PrependFontIdToFit` should work for any number of digits (it just doesn't do anything if the text already fits), but perhaps I'm wrong due to the alignment?

Reply:
Yeah the spaces fuck it all up, and when I remove the spaces and try to manually align, the text doesn't get overwritten if you change the number of digits mid transaction